### DISCUSSION TITLE
MP34DT05 PDM on ESP32

### DISCUSSION THREAD

**Discussion by Mukhin22** [pschatzmann/arduino-audio-tools#881] [Q&A] [answered]
Hey Phil!

Is it possible to use  #include "AudioLibs/AudioMP34DT05.h" like this example:
examples/examples-stream/streams-mp34dt05-serial/streams-mp34dt05-serial.ino
for esp32 not Arduino Nano?

Because when I'm trying to start it with my esp32 board on Arduino IDE "PDM.h" file used in the "AudioLibs/AudioMP34DT05.h" does not indexed properly.

Or can you recommend some other example of such a  PDM mic usage for esp32 with your beautiful libs

**Comment by Mukhin22** [comment]
I have tried with such configuration just with basic I2S api, but it doesn't seem to work - just noisy crackles. (double and triple checked all hw connections).

  info_micro.channels = 2;
  info_micro.sample_rate = 44100; // tried with 16000 - same
  info_micro.bits_per_sample = 16;
  /* Configure and start the input i2s */
  auto i2s_input_config = i2s_input.defaultConfig(RX_MODE);
  i2s_input_config.signal_type = audio_tools::PDM;
  i2s_input_config.copyFrom(info_micro);
  //i2s_input_config.i2s_format = I2S_STD_FORMAT;
  i2s_input_config.is_master = true;
  i2s_input_config.port_no = I2S_MICRO_PORT;
  i2s_input_config.pin_bck = I2S_MICRO_BCK_PIN;
  //i2s_input_config.pin_ws = I2S_MICRO_WS_PIN; 
  i2s_input_config.pin_data = I2S_MICRO_DATA_PIN;
  i2s_input.begin(i2s_input_config);

**Comment by pschatzmann** [comment]
I don't have any experience with the MP34DT05 PDM on the ESP32. 
You will really need to find some working examples using  and based on this we can determine the necessary settings to my API

**Reply by Mukhin22** [reply]
I think main problem is I'm trying to use PCB with ESP32-PICO-MINI-02. And there is no appropriate target for this board on Arduino IDE. Because I can't see the clock pulses both with ESP32-PICO-D4 and ESP32-WROOM-32DU targets with logic analyzer.
Do you know some workarounds how to run detect proper pin, if there is no target for such ESP32?

**Reply by Mukhin22** [reply]
What's about examples, I've found smth that should work https://github.com/RafigRzayev/Audio_Recording_to_WAV/blob/main/pdm_mic/pdm_mic.ino.
But I think I have to resolve proper target issue first.

**Comment by pschatzmann** [comment]
If I work with a processor which is not in the list I always use the ESP32 Dev Module.
If the example is working you will need to double check if it is still working if you change I2S_CHANNEL_FMT_ONLY_RIGHT to I2S_CHANNEL_FMT_RIGHT_LEFT since I don't support this parameter in your installed version.

**Comment by pschatzmann** [comment, marked as answer]
Here you can find some additonal info: https://esp32.com/viewtopic.php?t=6062
Looks like you need to define the channel_format and that the wiring (pin assignments) is a bit unexpected!

**Comment by pschatzmann** [comment]
I updated the documentation: https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC

**Reply by Mukhin22** [reply]
Thanks, I got it working more or less, part of the problem was hardware, now need to figure out if sound quality is good. It's not so easy because use case is micro+a2dp sound coming to speaker and they're very close to each other.
I'll check your documentation.
